### PR TITLE
[Components with more than 1 service] LPS-192612 Refactor ModelListener components part II

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-192612

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:analytics-cloud

**Comment by MrJohn1911** [comment]
ci:test:analytics-cloud-acceptance

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:analytics-cloud

**Comment by Kvale1733** [comment]
ci:test:startup

**Comment by Kvale1733** [comment]
ci:stop

**Comment by Kvale1733** [comment]
@MrJohn1911 There are unsatisfied references at startup
```
upgrade process com.liferay.adaptive.media.document.library.thumbnails.internal.upgrade.v1_0_0.DocumentLibraryThumbnailsConfigurationUpgradeProcess in 26 ms
[beanshell] 2023-08-09 19:02:09.273 INFO  [main][DependencyManagerSyncUtil:36] Available checkers: [Declarative Service Unsatisfied Component Checker, Spring Extender Unavailable Component Checker, Declarative Service Soft Circular Dependency Checker, Missing Component Scanner]
[beanshell] 2023-08-09 19:02:09.273 INFO  [main][DependencyManagerSyncUtil:62] Running "Declarative Service Unsatisfied Component Checker". You can run this by itself with command "ds:unsatisfied" in gogo shell.
[beanshell] 2023-08-09 19:02:16.907 WARN  [main][DependencyManagerSyncUtil:73] Declarative Service Unsatisfied Component Checker check result: 
[beanshell] Bundle {id: 1073, name: com.liferay.analytics.message.sender.impl, version: 5.0.35}
[beanshell] 	Declarative Service {id: 5012, name: com.liferay.analytics.message.sender.internal.model.listener.ExpandoRowModelListener, unsatisfied references: 
[beanshell] 		{name: _expandoRowEntityModelListener, target: (entity.model.listener.type=expandorow)}
```
[Log](https://testray.liferay.com/reports/production/logs/2023-08/test-1-29/test-portal-acceptance-pullrequest(master)/7821/functional-smoke-tomcat90-postgresql153-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:analytics-cloud

**Comment by MrJohn1911** [comment]
ci:test:analytics-cloud-acceptance

**Comment by Kvale1733** [comment]
ci:stop

**Comment by Kvale1733** [comment]
ci:test:analytics-cloud

**Comment by Kvale1733** [comment]
ci:test:startup

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:online-editing

**Comment by MrJohn1911** [comment]
ci:test:analytics-cloud

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:analytics-cloud

**Comment by MrJohn1911** [comment]
ci:test:analytics-cloud-acceptance

**Comment by catarineaguiar** [comment]
Analytics Cloud tests are broken. Let's wait for the fix to run the suites again https://github.com/brianchandotcom/liferay-portal/pull/139111

**Comment by zhangyan0701** [comment]
ci:test:analytics-cloud

**Comment by catarineaguiar** [comment]
ci:test:analytics-cloud-acceptance

**Comment by catarineaguiar** [comment]
ci:test:analytics-cloud

**Comment by catarineaguiar** [comment]
ci:test:analytics-cloud-acceptance

File: modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/model/listener/ContactModelListener.java [hunk 1]
```diff
@@ -7,36 +7,22 @@
...
 
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
 
 /**
  * @author Rachael Koestartyo
  */
-@Component(service = {EntityModelListener.class, ModelListener.class})
+@Component(service = ModelListener.class)
 public class ContactModelListener extends BaseAnalyticsModelListener<Contact> {
```
Comment (rafaprax) [inline modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/model/listener/ContactModelListener.java:22]: Hey @MrJohn1911, please rename these classes to `XyzAnalyticsModelListener`.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by Kvale1733** [comment]
ci:test:analytics-cloud

**Comment by catarineaguiar** [comment]
no regressions found

**Comment by MrJohn1911** [comment]
The last push was a component rename.

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by rafaprax** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
no regressions found

**Comment by dantewang** [comment]
@MrJohn1911 @rafaprax Does `BaseEntityModelListener` really need to be a `ModelListener`? The purpose of a model listener is to listen to CRUD events from our persistence layer.

After this PR, it seems `BaseEntityModelListener` and its sub-classes don't react to any CRUD event at all; instead, `BaseAnalyticsModelListener` and its sub-classes do. If this is the case, we should not implement `ModelListener` any more; we should neither call them `*EntityModelListener` any more.

If the `*EntityModelListener` classes need utilities from `com.liferay.analytics.message.sender.model.listener.BaseModelListener`, try to consider another way to put these utility methods.

Please resend, thanks!

cc @tinatian

**Pull request closed without merging**